### PR TITLE
Analytics Service added

### PR DESCRIPTION
**Pull request by paulk4633**

### CODE REVIEW

File: models/analytics.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	Order string `json:"order"`
+}
+
+type QueryResponse struct {
+	Data      Data_  `json:"data"`
+	RequestID string `json:"requestId"`
+	Status    string `json:"status"`
+}
+
+type Data_ struct {
```
Comment (Tigraine) [inline models/analytics.go]: `AnalyticsData` would be a better name

File: models/analytics.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	Result   Results    `json:"result"`
+	Messages []Messages `json:"messages"`
+}
+
+type Results struct {
```
Comment (Tigraine) [inline models/analytics.go]: `AnalyticsResult`

File: models/analytics.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	Rows         []interface{}  `json:"rows"`
+	RowCount     int32          `json:"rowCount"`
+	CollumnLabel []collumnLabel `json:"columnLabels"`
```
Comment (Tigraine) [inline models/analytics.go]: I'd delete this for now as it's not really necessary

File: models/analytics.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+type collumnLabel struct {
```
Comment (Tigraine) [inline models/analytics.go]: delete

File: services/rest_service.go [hunk 5]
```diff
@@ -26,6 +26,7 @@ func NewRestService(bitmovin *bitmovin.Bitmovin) *RestService {
 func (r *RestService) Create(relativeURL string, input []byte) ([]byte, error) {
 	fullURL := *r.Bitmovin.APIBaseURL + relativeURL
 	_, err := url.Parse(fullURL)
+
```
Comment (Tigraine) [inline services/rest_service.go]: remove blank line :)

File: models/analytics.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	Dimension  string    `json:"dimension"`
+	Start      string    `json:"start"`
+	End        string    `json:"end"`
+	Interval   *string   `json:"interval,omitempty"`
+	GroupBy    []string  `json:"groupBy,omitempty"`
+	Limit      *int32    `json:"limit,omitempty"`
+	Offset     *int32    `json:"offset,omitempty"`
+	LicenseKey *string   `json:"licenseKey,omitempty"`
+	Filters    []Filter  `json:"filters,omitempty"`
+	OrderBy    []OrderBy `json:"orderBy,omitempty"`
```
Comment (Tigraine) [inline models/analytics.go]: Missing `percentile` for the `percentile` function

File: examples/analytics/main.go [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+	if err != nil {
+		log.Printf("could not query %+v: %v", query, err)
+	}
+
+	log.Printf("%+v", res)
+
+}
+
+func stringToPtr(s string) *string {
```
Comment (peterpf) [inline examples/analytics/main.go]: remove unused functions

File: services/analytics_service.go [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+func (s *AnalyticsService) Max(a *models.Query) (*models.QueryResponse, error) {
+	return s.doAnalytics(a, "max")
+}
+
+func (s *AnalyticsService) Stddev(a *models.Query) (*models.QueryResponse, error) {
+	return s.doAnalytics(a, "stddev")
+}
+
+func (s *AnalyticsService) Percentile(a *models.PercentileQuery) (*models.QueryResponse, error) {
+	b, err := json.Marshal(*a)
```
Comment (peterpf) [inline services/analytics_service.go]: improve variable naming

**Pull request merged**